page-object/pages/DatasetsPage.ts

```diff
@@ -0,0 +1,36 @@
+
+import { Locator, Page } from "@playwright/test";
+import { BasePage } from "../BasePage";
+
+export class DatasetsPage extends BasePage{
+    datasetsPageLocator: any;
+    readonly upvoteButton: Locator;
+    readonly newNotebookButton: Locator;
+    readonly downloadButton: Locator;
+
+    constructor(page){
+        super(page);
+        this.datasetsPageLocator = page.locator("(//a[@class='sc-dmcoYd ggdQcj'])[4]");
+        this.upvoteButton = page.locator('//button[@data-testid="upvotebutton__upvote" and contains(@class, "sc-hSWyVn") and contains(@class, "sc-hEwMvu") and contains(@class, "iJoZbm") and contains(@class, "ekdYGr")]');
```

Comment:
A bit shorter Xpath for upvote button:
//button/preceding-sibling::button[@data-testid="upvotebutton__upvote"]

Reply:
thanks, fixed e739cebaa26a0e860d4beca624253b509ee4450a